_develop/connect/use-with-clarity.md

```diff
@@ -87,40 +124,104 @@ const functionArguments = [
-const myStatus = 'hey there';
-const options = {
-  contractAddress: 'ST22T6ZS7HVWEMZHHFK77H4GTNDTWNPQAX8WZAKHJ',
-  contractName: 'status',
-  functionName: 'write-status!',
-  functionArgs: [
-    {
-      type: 'buff',
-      value: myStatus,
-    }
-  ],
+// While in beta, you must provide this option:
+const authOrigin = 'https://deploy-preview-301--stacks-authenticator.netlify.app';
+
+openSTXTransfer({
+  recipient: 'ST2EB9WEQNR9P0K28D2DC352TM75YG3K0GT7V13CV',
+  amount: '100',
+  memo: 'Testing STX Transfers!',
+  authOrigin,
   appDetails: {
     name: 'SuperApp',
     icon: 'https://example.com/icon.png'
   },
-  finished: (data) => {
-    console.log('TX ID:', data.txId);
-    console.log('Raw TX:', data.txRaw);
+  finished: data => {
+    console.log(data.txId);
   },
-};
+});
+```
 
-await openContractCall(opts);
+When calling `openSTXTransfer`, you need to specify a few details. Here are the options you have:
+
+```ts
+interface STXTransferOptions {
+  recipient: string;
+  amount: string;
+  memo?: string;
+  authOrigin?: string;
+  appDetails: AuthOptions['appDetails'];
+  finished: (data: FinishedTxData) => void;
+}
 ```
 
-### Usage in React Apps
+parameter | type | optional | description
+---|---|---|---
+recipient | string | false | The STX Address for the recipient of this STX transfer
+amount | string | false | The amount of microstacks (µSTX) to be transferred. This argument is a string to prevent floating point errors. There are 1,000,000 µSTX per STX.
+memo | string | true | An optional memo to include in the transaction.
+appDetails | object | false | A dictionary that includes `name` and `icon`
+finished | function | false | A callback that is fired when the transaction is signed and broadcasted. Your callback will receive an object back with a `txId` and a `txRaw`, both of which are strings.
+authOrigin | string | true | The location of the authenticator. This is only necessary when developing the authenticator locally, or when using beta features. Defaults to `"https://app.blockstack.org"`.
+
+## Deploying Clarity Contracts
+
+To allow your app's users to deploy arbitrary Clarity contracts, use the `openContractDeploy` method.
+
+```ts
+import { openContractDeploy } from '@blockstack/connect';
+
+const contractSource = '(begin (print "hello, world"))';
+// While in beta, you must provide this option:
+const authOrigin = 'https://deploy-preview-301--stacks-authenticator.netlify.app';
+
+openContractDeploy({
+  contractName: 'my-contract-name',
+  contractSource,
+  authOrigin,
+  appDetails: {
+    name: 'SuperApp',
+    icon: 'https://example.com/icon.png'
+  },
+  finished: (data) => {
+    console.log(data.txId);
+  }
+})
+```
+
+Here is the interface for the options you can provide to `openContractDeploy`:
+
+```ts
+interface ContractDeployOptions {
+  contractSource: string;
+  contractName: string;
+  authOrigin?: string;
+  appDetails: AuthOptions['appDetails'];
+  finished: (data: FinishedTxData) => void;
+}
```

Comment:
Naming conflict between the original options for the underlying function. Does it matter if one is `codeBody` and the other is `contractSource`?

```tsx
export interface ContractDeployOptions {
  contractName: string;
  codeBody: string;
  senderKey: string;
  fee?: BigNum;
  nonce?: BigNum;
  network?: StacksNetwork;
  anchorMode?: AnchorMode;
  postConditionMode?: PostConditionMode;
  postConditions?: PostCondition[];
}
```

Also, could we allow the types for `openContractDeploy` to be `Partial<ContractDeployOptions>` from the stacks-transactions lib?

Reply:
I updated the argument to be `codeBody` instead, and have [this issue](https://github.com/blockstack/ux/issues/408) to move to API parity with stacks-transactions.